python/ray/util/sgd/torch/torch_dataset.py

```diff
@@ -0,0 +1,172 @@
+import functools
+import logging
+from collections import Iterator
+from collections.abc import Iterable
+from typing import Any, Callable, List, Optional
+
+import numpy as np
+import torch
+import pandas as pd
+from torch.utils.data import IterableDataset
+
+from ray.util.data import MLDataset, TensorDataset
```

Comment:
What's the difference between a MLDataset and TensorDataset?

Reply:
`MLDataset ` can be treated as a special version of ParallelIterator[pandas.DataFrame]. TensorDataset is a common interface for `TorchMLDataset` and `TFMLDataset`. It has limited function and only used for torch or tf.